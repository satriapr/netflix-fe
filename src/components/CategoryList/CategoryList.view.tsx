import React from 'react'

import Constant from '../../constants/Constant'
import { IMovieReq } from '../../models'
import Category from '../Category/Category.view'

// OMDB API can't retrieve 'type'(genre) without keyword, so i harcode the keyword.
const movies: IMovieReq[] = [
  {
    keyword: Constant.MAN,
    type: Constant.MOVIE,
    categoryTitle: Constant.POPULAR_ON_NETFLIX
  },
  {
    keyword: Constant.GUN,
    type: Constant.MOVIE,
    categoryTitle: Constant.CLASSIC_FILM
  },
  {
    keyword: Constant.MAN,
    type: Constant.SERIES,
    categoryTitle: Constant.TV_SERIES
  }
]

const CategoryList = () => {
  return (
    <React.Fragment>
      {movies.map((movie: IMovieReq, i) => {
        const { keyword, type, categoryTitle } = movie

        return (
          <div key={i}>
            <Category
              keyword={keyword}
              type={type}
              categoryTitle={categoryTitle}
            />
          </div>
        )
      })}
    </React.Fragment>
  )
}

export default CategoryList
