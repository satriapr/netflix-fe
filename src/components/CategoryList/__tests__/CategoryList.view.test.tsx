import CategoryList from '../CategoryList.view'
import { testSnapshots } from '../../../utils/test.util'

describe('CategoryList Component', () => {
  testSnapshots(CategoryList, [
    {
      description: 'CategoryList props',
      props: {}
    }
  ])
})
