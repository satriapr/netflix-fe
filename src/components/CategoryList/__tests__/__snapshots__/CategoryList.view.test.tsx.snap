// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList Component snapshots CategoryList props 1`] = `
<Fragment>
  <div
    key="0"
  >
    <Category
      categoryTitle="Popular on Netflix"
      keyword="man"
      type="movie"
    />
  </div>
  <div
    key="1"
  >
    <Category
      categoryTitle="Classic Film"
      keyword="gun"
      type="movie"
    />
  </div>
  <div
    key="2"
  >
    <Category
      categoryTitle="TV Series"
      keyword="man"
      type="series"
    />
  </div>
</Fragment>
`;
